app/views/ProjectEdit/Connector/EditConnectorModal/index.tsx

```diff
@@ -0,0 +1,275 @@
+                    alert.show(
+                        'Successfully created a new connector.',
+                        { variant: 'success' },
+                    );
+                }
+            },
+            onError: (errors) => {
+                setError({
+                    [internal]: errors.message,
+                });
+            },
+        },
+    );
+
+    const {
+        setValue: onRowChange,
+    } = useFormArray('sources', setFieldValue);
+
+    const error = getErrorObject(riskyError);
+
+    const handleSourceClick = useCallback((key: SupportedSource) => {
+        setFieldValue(
+            (oldSources: PartialSourceType[] | undefined = []) => {
+                const selectedSourceIndex = oldSources
+                    ?.findIndex((source) => source.source === key);
+
+                if (selectedSourceIndex === -1) {
+                    return [
+                        ...(oldSources ?? []),
+                        {
+                            clientId: randomString(),
+                            title: sourcesLabel[key],
+                            params: {},
+                            source: key,
+                        },
+                    ];
+                }
+                return oldSources?.filter((source) => source.source !== key);
+            },
+            'sources' as const,
+        );
+    }, [setFieldValue]);
+
+    const usedSourceMap = listToMap(
+        value?.sources?.filter((source) => isDefined(source.source)),
+        (source) => (source.source),
+        () => true,
+    );
+
+    const buttonRendererParams = useCallback((key: SupportedSource) => ({
+        name: key,
+        value: !!usedSourceMap?.[key],
+        onClick: handleSourceClick,
+        children: sourcesLabel[key] ?? key,
+    }), [
+        usedSourceMap,
+        handleSourceClick,
+    ]);
+
+    const sourcesError = getErrorObject(error?.sources);
+
+    const connectorSourceRendererParams = useCallback((
+        key: string,
+        data: PartialSourceType,
+        index,
+    ) => ({
+        name: index,
+        error: sourcesError?.[key],
+        value: data,
+        onChange: onRowChange,
+    }), [onRowChange, sourcesError]);
+
+    const handleSubmit = useCallback(() => {
+        const submit = createSubmitHandler(
+            validate,
+            setError,
+            (finalValue) => {
+                createConnector({
+                    variables: {
+                        projectId,
+                        input: finalValue as ConnectorInputType,
+                    },
+                });
+            },
+        );
+        submit();
+    }, [
+        projectId,
+        setError,
+        validate,
+        createConnector,
+    ]);
+
+    return (
+        <Modal
+            className={_cs(styles.editConnectorModal, className)}
+            onCloseButtonClick={onClose}
+            size="cover"
+            heading={connectorId ?? 'Add New Connector'}
```

Comment:
Huh?

Reply:
This is in progress. I'll add a note

Reply:
Let's just have a "Edit" or "New" connector title